darts/explainability/shap_explainer.py

```diff
@@ -698,13 +700,33 @@ def _create_regression_model_shap_X(
             past_covariates=past_covariates,
             future_covariates=future_covariates,
             lags=lags_list,
+            concatenate=False,
             lags_past_covariates=lags_past_covariates_list if past_covariates else None,
             lags_future_covariates=lags_future_covariates_list
             if future_covariates
             else None,
         )
         # Remove sample axis:
-        X = X[:, :, 0]
+        for i, X_i in enumerate(X):
+            X[i] = X_i[:, :, 0]
```

Comment:
I would rather keep the first implementation, without the loop.

Reply:
You're indeed correct, I overlooked the multi-timeseries scenario.

It's alright like this!